pkg/kubefed2/join.go

```diff
@@ -81,6 +82,8 @@ func (o *joinFederationOptions) Bind(flags *pflag.FlagSet) {
 		"Name of the secret where the cluster's credentials will be stored in the host cluster. This name should be a valid RFC 1035 label. If unspecified, defaults to a generated name containing the cluster name.")
 	flags.BoolVar(&o.addToRegistry, "add-to-registry", false,
 		"Add the cluster to the cluster registry that is aggregated with the kubernetes API server running in the host cluster context.")
+	flags.BoolVar(&o.limitedScope, "limited-scope", false,
+		"Whether the federation namespace will be the only target for federation.  If true, join will add a service account with access only to the federation namespace in the target cluster.")
```

Comment:
Consider adding a reference to the `--federation-namespace` flag to this flag description.

Also, the description for the `federation-namespace` flag probably needs to be updated to reflect that it can be the namespace to target in the joining cluster.

Reply:
Done.

The thought exercise of considering the relationship between the 2 args led to the thought that `kubefed2` should be discovering control plane configuration (the cluster registry namespace and whether the control plane is targeting a single namespace) rather than requiring it to be provided on every invocation.

Reply:
Yes, agreed! Perhaps that same mechanism can be used to configure the federation control plane.